src/trigger/TriggerGenericMaker.hpp

```diff
@@ -78,36 +124,178 @@ class TriggerGenericMaker : public dunedaq::appfwk::DAQModule
     while (running_flag.load()) {
       IN in;
       try {
-        m_input_queue->pop(in, m_queue_timeout);
+        parent.m_input_queue->pop(in, parent.m_queue_timeout);
       } catch (const dunedaq::appfwk::QueueTimeoutExpired& excpt) {
         // it is perfectly reasonable that there might be no data in the queue
         // some fraction of the times that we check, so we just continue on and try again
         continue;
       }
       ++received_count;
-
+  
       std::vector<OUT> outs;
-      m_maker->operator()(in, outs);
-
-      while (outs.size()) {
-        bool successfullyWasSent = false;
-        while (!successfullyWasSent && running_flag.load()) {
-          try {
-            m_output_queue->push(outs.back(), m_queue_timeout);
-            outs.pop_back();
-            successfullyWasSent = true;
-            ++sent_count;
-          } catch (const dunedaq::appfwk::QueueTimeoutExpired& excpt) {
-            ers::warning(excpt);
+      try {
+        parent.m_maker->operator()(in, outs);
+      } catch (...) {
+        ers::error(AlgorithmFatalError(ERS_HERE, parent.get_name(), "algorithm_name_placeholder"));
+        continue;
```

Comment:
If the algorithm throws, I think we should assume it's not going to work any more, and so we should `break` rather than `continue`.  We can then make the ers call an `ers::fatal` (which just increases the severity of the message; it doesn't exit the program like in other logging systems).